### PR TITLE
Fix8

### LINKED ISSUE #8: Invalid parameter values in [dpqr] functions should return NaN
Make invalid parameters return NaN, instead than stoping. Done in some functions, check in all.

### PR DESCRIPTION
**Pull request by andrechalom**
Completely reworked the [d] and [p] functions to produce NaN instead of stopping at invalid values. This is a partial fix for #8 (as the [q] functions are not affected yet. [q] functions will probably be fixed along with work on issue #35)
Edit: now fixes #8 and #35

### CODE REVIEW

File: R/dbs.R [hunk 1]
```diff
@@ -1,10 +1,9 @@
 dbs <- function(x, N, S, log = FALSE){
-  if (!all(is.finite(c(N, S)))) stop("all parameters should be finite")
-  if (N <= 0)  stop("N must be larger than zero")
-  if (S <= 0)  stop("S must be larger than zero")
-  if(any(x < 1)) stop("at least one x less than one")
-  if(any(!is.wholenumber(x))) stop("all x must be integers")
-  y <- (S-1)*(1-x/N)^(S-2)/N
-  if(log) return (log(y))
-  else return(y)
+	x[ ! is.wholenumber(x) | x < 1] <- NaN
```
Comment (piklprado) [inline R/dbs.R]: At least for x<1 should'nt return zero. See may comments below.

**Comment by piklprado** [comment]
Wow, impressive job! Just a question: the idea here is to stick to the standard for density functions in R:
1 - For invalid parameter values they return NaN. Among other things this is assumed by nle and mle2 functions. OK
2 - But for variate values outside the sample sapce (aka 'support') of the function the value returned is zero, not NaN. For example:
```r
> dpois(-1, 0.01)
[1] 0
>  dlnorm(-1)
[1] 0
```
But for `dgs` for example we got:

```
> dgs(-1, k=1, S=10)
[1] NaN
Mensagens de aviso perdidas:
In dgs(-1, k = 1, S = 10) : NaNs produced
```

File: R/dbs.R [hunk 2]
```diff
@@ -1,10 +1,9 @@
 dbs <- function(x, N, S, log = FALSE){
-  if (!all(is.finite(c(N, S)))) stop("all parameters should be finite")
-  if (N <= 0)  stop("N must be larger than zero")
-  if (S <= 0)  stop("S must be larger than zero")
-  if(any(x < 1)) stop("at least one x less than one")
```
Comment (piklprado) [inline R/dbs.R]: The current version makes also returns an error when x is outside the sample space. My fault, sorry.

**Comment by andrechalom** [comment]
Right, I'll work on that.

**Comment by andrechalom** [comment]
Now, values of x that are not in the function support will return 0, and non-integer values throw a warning if the function only accepts integer values for x:

```R
> dbs(0:4/2, 40, 50)
[1] 0.0000000 0.0000000 0.3633772 0.0000000 0.1044405
Warning message:
In dbs(0:4/2, 40, 50) : non integer values in x
```

**Comment by andrechalom** [comment]
Also: fixing zipf distribution to return 0 for x > N.

**Pull request merged**